### DISCUSSION TITLE
Stack para desenvolvimento front end atual

### DISCUSSION THREAD

**Discussion by wellingtonpgp** [frontendbr/forum#535] [Perguntas e Respostas]
Galera qual stack estão usando atualmente para front-end​? 😀

**Comment by ghost** [comment]
Fala @wellingtongeek , seguinte cara:

**Sites simples:** html, css e js com automatizações no gulp
**Sistemas mais complexo:** Avalio a necessidade e tento aplicar a tecnologia para o universo do sistemas, por exemplo estou atualmente trabalhando com um sistema escrito em elixir e no frontend estou usando React, Redux, Webpack e Storybook. Ai adicionei alguns scripts para automatização.

Ai se a dúvida é qual usar analise a situação, se js puro ou um jquery resolve então vá por essa abordagem hehe.

**Comment by caioincau** [comment]
Estou trabalhando hoje com Webpack, Vue, Vuex e BrowserSync, ao mesmo tempo toco outro projeto menor com JQuery e Gulp, vai do tipo de projeto.

**Comment by mahenrique94** [comment]
Muitos usam React, Angular2, Vue, MEAN Stack 2, etc... Varia da sua necessidade e escopo do projeto.

**Comment by fabioassuncao** [comment]
Cara, uso o gulp com alguns plugins 
(concat, cssmin, rename, uglify, etc) em projetos atuais, mas estou pensando muito em usar o Webpack para as novas demandas.

**Comment by matheusml** [comment]
Atualmente estou usando:

- React
- Webpack 2
- ES2015
- CSS Modules (querendo migrar para [styled-components](https://styled-components.com/))
- Mocha (querendo migrar para [Jest](https://facebook.github.io/jest/))
- ESLint

Gosto bastante desse _stack_, principalmente para projetos pequenos/médios.

**Comment by kvnol** [comment]
Como não trabalho com sistemas complexos ainda, minha stack está sendo:

- Pug
- SASS (utilizando atomic design)
- Gulp

**Comment by Cleysonlb** [comment]
Front End
  - Gulp (com diversos plugins)
  - Stylus (pré processador)
  - Jeet (grid system)
  - Rupture (media querie system)
  - Stylint
  - JSlint

**Comment by kazzkiq** [comment]
Pra projetos pessoais ou freelas eu geralmente uso:

- [Brunch](https://github.com/brunch/brunch) (build system)
- [Riot.js](https://github.com/riot/riot) (lib de componentização)
- [SASS](https://github.com/sass/sass)
- [Viewports](https://github.com/jareware/viewports) (media queries)
- [Babel](https://github.com/babel/babel) (pra poder codar em ES6/ES7)
- [Jest](https://github.com/facebook/jest) (comecei a aprender e aplicar TDD/BDD, inclusive recomendo muito o [tutorial](http://blog.da2k.com.br/2015/01/06/pensando-tdd-com-javascript/) do @fdaciuk)

O bacana desse stack é que a quantidade de configuração é quase zero, as libs são super simples de usar, e quase tudo funciona *out-of-the-box*, o que ajuda muito na produtividade.

Já pra projetos no trabalho é quase sempre unanimidade (e as vezes vem como pedido do próprio cliente) o uso das tecnologias já consolidadas no mercado como Angular (1 ou 2), React, Redux, Webpack, Vue.js, etc.

**Comment by devigor** [comment]
#### Projetos Simples
  - Gulp
  - Pug
  - Stylus :heart: 
  - Js (ES6/ES6) :heart:

#### SPA's
  - React
  - Redux
  - Webpack
  - MaterialUI

**Comment by devigor** [comment]
Lembrando que vai da necessidade de cada projeto :smile:

**Comment by alanhr** [comment]
- React 
- Redux
- flowtype
- Es6
- Webpack
- Eslint
- CSS Module
- Mocha
